views/utils.py

```diff
@@ -112,10 +86,44 @@ def _load_challenges() -> dict[ChallengeName, Challenge]:
         return challenges
 
     @staticmethod
-    def _type_check_with_mypy(code) -> TypeCheckResult:
+    def _type_check_with_mypy(code: str) -> TypeCheckResult:
+        buffer = io.StringIO(code)
+        tokens = list(tokenize.generate_tokens(buffer.readline))
```

Comment:
Let's add a comment that points to https://docs.python.org/3.12/library/tokenize.html#tokenize.tokenize, since it's really hard to understand the output format without reading the doc 😂